feat(curriculum): add second set of quiz questions for Responsive Web Design

saif814 opened the pull request:
feat(certification): add second set of quiz questions for Responsive Web Design

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or Gitpod.

Closes #59754

This PR adds the second set of quiz questions to the Responsive Web Design certification module. All questions have been reviewed for clarity and correctness.

Commit:
Added Set 2 of Responsive Web Design quiz questions

All questions have been added and reviewed.

saif814 commented:
Hi! I've updated the PR description and followed the contribution and PR opening guidelines. Just waiting for review now. Please let me know if anything needs changing. Thanks!

saif814 commented:
@camper-bot please update

ilenia-magoni commented:
@saif814 what do you want the bot to do? your PR still doesn't respect the guidelines

saif814 commented:
> @saif814 what do you want the bot to do? your PR still doesn't respect the guidelines

sorry but i changed the format of the PR and i am really confused that why this does not follow the guidelines can you help ??

ilenia-magoni commented:
@saif814 you did not change the title of the PR

saif814 commented:
> feat(certification): add second set of quiz questions for Responsive Web Design

feat(certification): add second set of quiz questions for Responsive Web Design this is the new title of the PR right its different from the previous one this is my first PR i have made on github so i might be wrong but can you help me by telling exactly whats wrong

ilenia-magoni commented:
@saif814 you added that in the body of the PR, you have not changed the title

ilenia-magoni commented:
@saif814 you have not changed the PR title
![image](https://github.com/user-attachments/assets/cd5feb4d-3670-41a8-98c4-c243a16247c9)

saif814 commented:
> @saif814 you have not changed the PR title ![image](https://private-user-images.githubusercontent.com/26656284/435775074-cd5feb4d-3670-41a8-98c4-c243a16247c9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUyNjAwNTAsIm5iZiI6MTc0NTI1OTc1MCwicGF0aCI6Ii8yNjY1NjI4NC80MzU3NzUwNzQtY2Q1ZmViNGQtMzY3MC00MWE4LTk4YzQtYzI0M2ExNjI0N2M5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDIxVDE4MjIzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc3NjkxMTAyZjI2NjQ4ZjcyZjUwNDc4NjUyZmVjOTlhMGVmNDU4NDYxNWNjNWZmZmEyYmIxNTgyZGNhMTNmZmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3skrN-acH7a6tsuEZJ5-D3qUk75Ozy2X1mmsnJ_ddVg)

ohhh hey does it need to be same as 

feat(certification): add second set of quiz questions for Responsive Web Design   

this ?

ilenia-magoni commented:
@saif814 yes, that should be the title, almost that, `certification` is not one of the scopes that are available

saif814 commented:
> @saif814 yes, that should be the title, almost that, `certification` is not one of the scopes that are available

feat(curriculum): add second set of quiz questions for Responsive Web Design

how about this ?

ilenia-magoni commented:
great, thank you

ilenia-magoni commented:
>  ValidationError: "quizzes[1].questions[6].distractors[1]" contains an excluded value

your seventh question has an issue, specifically the second distractor

Commit:
fix: corrected distractor for question 17

I’ve corrected the second distractor on question 17 to avoid confusion with the correct answer

saif814 commented:
Thanks a lot for pointing that out! I really appreciate your feedback. Let me know if there’s anything else you'd suggest.

ilenia-magoni commented:
the question with issues is not the seventeenth, it's the seventh, ten before the one you are changing where one distractor is equal to the answer

Commit:
fix(curriculum): corrected question 7 wording in quiz

2nd option for the distractors have been changed.

zairahira commented on curriculum/challenges/english/25-front-end-development/quiz-responsive-web-design/66ed9034f45ce3ece4053ebd.md:
```diff
@@ -523,3 +523,444 @@ What is the default media type if no media type is specified?
+
+#### --text--
+
+What is the primary purpose of using media queries in responsive web design?
```
This is a duplicate of question no.2 in this set.

zairahira commented on curriculum/challenges/english/25-front-end-development/quiz-responsive-web-design/66ed9034f45ce3ece4053ebd.md:
```diff
@@ -523,3 +523,444 @@ What is the default media type if no media type is specified?
+
+---
+
+`max-width`
```
This distractor needs to be changed as `max-width` is also a valid option.

zairahira commented on curriculum/challenges/english/25-front-end-development/quiz-responsive-web-design/66ed9034f45ce3ece4053ebd.md:
```diff
@@ -523,3 +523,444 @@ What is the default media type if no media type is specified?
+
+#### --text--
+
+What is the purpose of the `@media (hover: hover)` query in CSS?
```
This question is same as the one below:

```
What does the `@media (hover: hover)` media query check for?
```

jdwilkin4 replied:
@saif814 

was this comment addressed?

saif814 replied:
yes i did fixed that i replace that with a new one

Commit:
Update 66ed9034f45ce3ece4053ebd.md

Refreshed quiz with unique questions to keep learning engaging and non-repetitive

Commit:
Update 66ed9034f45ce3ece4053ebd.md

corrected format

saif814 commented on curriculum/challenges/english/25-front-end-development/quiz-responsive-web-design/66ed9034f45ce3ece4053ebd.md:
```diff
@@ -523,3 +523,444 @@ What is the default media type if no media type is specified?
+
+`@media screen and (min-resolution: 300dpi)`
+
+--question--
```
i have corrected the formatting can you please recheck

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/quiz-responsive-web-design/66ed9034f45ce3ece4053ebd.md:
```diff
@@ -523,3 +523,444 @@ What is the default media type if no media type is specified?
+
+`@media screen and (min-resolution: 300dpi)`
+
+### --question--
```
use backticks around the distractors and answer of this question

saif814 replied:
it done

Commit:
Update 66ed9034f45ce3ece4053ebd.md

added backtics

zairahira commented on curriculum/challenges/english/25-front-end-development/quiz-responsive-web-design/66ed9034f45ce3ece4053ebd.md:
```diff
@@ -771,23 +771,23 @@ Which media feature allows you to apply styles only on high pixel density screen
 #### --text--
 
-What does the `@media (hover: hover)` media query check for?
+What does the pointer media feature detect in a media query?
```
This topic isn't discussed in the responsive web design section.

saif814 replied:
are you pointing out the question that says pointer media feature detect or the one that goes like `@media (hover: hover)` media query check for ?

zairahira replied:
Yes

saif814 replied:
i have replace the ( what does the pointer media feature detect in a media qyery ) with a new question

Commit:
Update 66ed9034f45ce3ece4053ebd.md

replaced with a new question

zairahira commented on curriculum/challenges/english/25-front-end-development/quiz-responsive-web-design/66ed9034f45ce3ece4053ebd.md:
```diff
@@ -523,3 +523,444 @@ What is the default media type if no media type is specified?
+
+#### --text--
+
+What is the primary purpose of a media query in responsive web design?
```
This question is same as this one:

https://github.com/freeCodeCamp/freeCodeCamp/blob/1303bc3af8dd2bae2c8b15efd942c91531c88598/curriculum/challenges/english/25-front-end-development/quiz-responsive-web-design/66ed9034f45ce3ece4053ebd.md?plain=1#L554

saif814 replied:
this was replaced with the new commit i just made

zairahira commented on curriculum/challenges/english/25-front-end-development/quiz-responsive-web-design/66ed9034f45ce3ece4053ebd.md:
```diff
@@ -523,3 +523,444 @@ What is the default media type if no media type is specified?
+
+---
+
+It detects if the device supports a dark theme.
```
This seems out of context.

```suggestion
It detects if the device uses a touchscreen.
```

saif814 replied:
this has been applied

zairahira commented on curriculum/challenges/english/25-front-end-development/quiz-responsive-web-design/66ed9034f45ce3ece4053ebd.md:
```diff
@@ -523,3 +523,444 @@ What is the default media type if no media type is specified?
+
+---
+
+Setting styles in JavaScript based on time of day
```
This distractor seems out of context.
```suggestion
Using JavaScript to detect screen size and apply classes.
```

saif814 replied:
this change was also applied

zairahira commented on curriculum/challenges/english/25-front-end-development/quiz-responsive-web-design/66ed9034f45ce3ece4053ebd.md:
```diff
@@ -523,3 +523,444 @@ What is the default media type if no media type is specified?
+
+#### --text--
+
+Which media query feature is used to detect the screen orientation (landscape or portrait)?
```
This question is very similar to:

https://github.com/freeCodeCamp/freeCodeCamp/blob/1303bc3af8dd2bae2c8b15efd942c91531c88598/curriculum/challenges/english/25-front-end-development/quiz-responsive-web-design/66ed9034f45ce3ece4053ebd.md?plain=1#L818

saif814 replied:
one of these were replace by a new question

zairahira reviewed (changes requested):
Hey, thanks for applying the suggestions I left earlier.
I have a few more suggestions before we can get this in.

Commit:
Update 66ed9034f45ce3ece4053ebd.md

with suggested changes and replacements.

zairahira commented on curriculum/challenges/english/25-front-end-development/quiz-responsive-web-design/66ed9034f45ce3ece4053ebd.md:
```diff
@@ -523,3 +523,444 @@ What is the default media type if no media type is specified?
+
+#### --answer--
+
+`prefers-reduced-motion`
```
This topic is taught in the CSS Animations section that comes later in the course. Kindly replace this question with a topic from the current section (Responsive Webdesign) only.

zairahira reviewed (changes requested):
Everything is looking good, I have just one more suggestion.

Commit:
Update 66ed9034f45ce3ece4053ebd.md

with added suggestions

Commit:
Update 66ed9034f45ce3ece4053ebd.md

with suggested changes applied

saif814 commented on curriculum/challenges/english/25-front-end-development/quiz-responsive-web-design/66ed9034f45ce3ece4053ebd.md:
```diff
@@ -573,23 +573,23 @@ To adjust the layout and design based on screen size
 #### --text--
 
-Which media feature can be used to detect if a user prefers reduced motion for accessibility reasons?
+Which CSS value allows a container to adjust its width automatically based on the size of its parent in responsive design?
```
yes i have applied your suggested change.

jdwilkin4 commented:
Congrats on your first contribution to our repository. 🎉
We look forward to future contributions. 👍
